Index: pom.xml
===================================================================
--- pom.xml	(revision 742)
+++ pom.xml	(working copy)
@@ -52,6 +52,12 @@
 			<groupId>se.repos</groupId>
 			<artifactId>repos-indexing</artifactId>
 			<version>0.9.2</version>
+			<exclusions>
+				<exclusion>
+					<artifactId>jsr311-api</artifactId>
+					<groupId>javax.ws.rs</groupId>
+				</exclusion>
+			</exclusions>
 		</dependency>	
 		<dependency>
 			<groupId>se.repos</groupId>
@@ -74,7 +80,19 @@
 			<groupId>se.simonsoft</groupId>
 			<artifactId>cms-indexing-xml</artifactId>
 			<version>0.14-SNAPSHOT</version>
+			<exclusions>
+				<!-- using JAX-RS 2 (Client) in this module -->
+				<exclusion>
+					<groupId>javax.ws.rs</groupId>
+					<artifactId>jsr311-api</artifactId>
+				</exclusion>
+			</exclusions>
 		</dependency>
+ 		<dependency>
+			<groupId>se.repos</groupId>
+			<artifactId>repos-indexing-graphdb</artifactId>
+			<version>1-SNAPSHOT</version>
+		</dependency>		
 		<dependency>
 			<groupId>se.simonsoft</groupId>
 			<artifactId>cms-backend-svnkit</artifactId>
@@ -118,6 +136,18 @@
 	</dependencies>
 	
 	<build>
+		<pluginManagement>
+			<plugins>
+				<plugin>
+					<artifactId>maven-compiler-plugin</artifactId>
+					<configuration>
+						<source>1.7</source>
+						<target>1.7</target>
+						<encoding>UTF-8</encoding>
+					</configuration>
+				</plugin>
+			</plugins>
+		</pluginManagement>
 		<plugins>
 			<plugin>
 				<artifactId>maven-assembly-plugin</artifactId>
Index: src/main/java/se/repos/indexing/standalone/config/IndexingHandlersModuleXml.java
===================================================================
--- src/main/java/se/repos/indexing/standalone/config/IndexingHandlersModuleXml.java	(revision 742)
+++ src/main/java/se/repos/indexing/standalone/config/IndexingHandlersModuleXml.java	(working copy)
@@ -5,6 +5,8 @@
 
 import se.repos.indexing.IndexingItemHandler;
 import se.repos.indexing.fulltext.HandlerFulltext;
+import se.repos.indexing.graphdb.neo4j.Neo4jIndexedAnalysis;
+import se.repos.indexing.graphdb.neo4j.Neo4jIndexingItemHandler;
 import se.simonsoft.cms.indexing.xml.IndexAdminXml;
 import se.simonsoft.cms.indexing.xml.IndexingHandlersXml;
 import se.simonsoft.cms.indexing.xml.XmlIndexFieldExtraction;
@@ -35,6 +37,10 @@
 		Multibinder<XmlIndexFieldExtraction> xmlExtraction = Multibinder.newSetBinder(binder(), XmlIndexFieldExtraction.class);
 		IndexingHandlersXml.configureXmlFieldExtraction(xmlExtraction);
 		
+		handlers.addBinding().to(Neo4jIndexingItemHandler.class);
+		xmlExtraction.addBinding().to(Neo4jIndexingItemHandler.class);
+		handlers.addBinding().to(Neo4jIndexedAnalysis.class);
+		
 		bind(XmlMatchingFieldExtractionSource.class).to(XmlMatchingFieldExtractionSourceDefault.class);
 	}
 
Index: src/main/java/se/repos/indexing/standalone/config/ParentModule.java
===================================================================
--- src/main/java/se/repos/indexing/standalone/config/ParentModule.java	(revision 742)
+++ src/main/java/se/repos/indexing/standalone/config/ParentModule.java	(working copy)
@@ -3,9 +3,20 @@
  */
 package se.repos.indexing.standalone.config;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.ws.rs.client.WebTarget;
+
 import org.apache.solr.client.solrj.SolrServer;
 import org.tmatesoft.svn.core.wc.admin.SVNLookClient;
 
+import se.repos.indexing.graphdb.neo4j.CypherTransaction;
+import se.repos.indexing.graphdb.neo4j.Neo4jClientJaxrsProvider;
+import se.repos.indexing.graphdb.neo4j.Neo4jIndices;
+import se.repos.indexing.graphdb.neo4j.Neo4jIndicesCypherConstraints;
+import se.repos.indexing.graphdb.neo4j.tx.CypherTransactionProviderRestJsonBatch;
 import se.repos.indexing.scheduling.IndexingSchedule;
 import se.repos.indexing.scheduling.IndexingScheduleBlockingOnly;
 import se.simonsoft.cms.backend.svnkit.svnlook.CmsRepositoryLookupSvnkitLook;
@@ -13,6 +24,7 @@
 import se.simonsoft.cms.item.info.CmsRepositoryLookup;
 
 import com.google.inject.AbstractModule;
+import com.google.inject.TypeLiteral;
 import com.google.inject.name.Names;
 
 public class ParentModule extends AbstractModule {
@@ -32,6 +44,21 @@
 		// this is an all-inspection context
 		bind(SVNLookClient.class).toProvider(SvnlookClientProviderStateless.class);
 		bind(CmsRepositoryLookup.class).to(CmsRepositoryLookupSvnkitLook.class);
+		
+		// authoring stuff
+		bind(new TypeLiteral<Set<String>>(){}).annotatedWith(Names.named("config:se.repos.indexing.authoringUnitElements")).toInstance(new HashSet<String>(Arrays.asList(
+				// techdoc
+				"p", "title", "figuretext",
+				// checksheets
+				"title", "description", "summary", "reply"
+				)));
+		// graphdb
+		bind(WebTarget.class).annotatedWith(Names.named("neo4j")).toProvider(Neo4jClientJaxrsProvider.class); // TODO as singleton?
+		bind(CypherTransaction.class).toProvider(CypherTransactionProviderRestJsonBatch.class);
+		// initialize constraints and indexes
+		bind(Neo4jIndices.class).to(Neo4jIndicesCypherConstraints.class)
+			// at startup
+			.asEagerSingleton();
 	}
 	
 }
Index: src/main/resources/log4j2.xml
===================================================================
--- src/main/resources/log4j2.xml	(revision 742)
+++ src/main/resources/log4j2.xml	(working copy)
@@ -14,6 +14,20 @@
 		</Console>
 	</Appenders>
 	<Loggers>
+		<!-- neo4j dev -->
+		<Logger name="se.repos.indexing.graphdb" level="trace" additivity="false">
+			<AppenderRef ref="Console"/>
+		</Logger>			
+		<Logger name="se.repos.indexing" level="info" additivity="false">
+			<AppenderRef ref="Console"/>
+		</Logger>
+		<Logger name="se.simonsoft.cms.backend" level="warn" additivity="false">
+			<AppenderRef ref="Console"/>
+		</Logger>
+		<Logger name="se.simonsoft.cms.indexing" level="info" additivity="false">
+			<AppenderRef ref="Console"/>
+		</Logger>		
+		<!-- end neo4j dev -->
 		<Root level="${level}">
 			<AppenderRef ref="Console" />
 		</Root>
